tests/unit/writers/qbwriter/test_dsdtordfmodelshelper.py

```diff
@@ -577,5 +578,69 @@ def test_qb_order_of_components():
     ) in graph
 
 
+def test_units_component_duplication():
+    """This test checks if there are multiple observation value columns with units, only one unit component will be generated.
+    link to the issue: https://github.com/GSS-Cogs/csvcubed/issues/755
+    """
+
+    data = pd.DataFrame(
+        {
+            "Existing Dimension": ["A", "B", "C"],
+            "Value": [1, 2, 3],
+            "Existing Attribute": ["Provisional", "Final", "Provisional"],
+        }
+    )
+
+    cube = Cube(
+        CatalogMetadata("Some Dataset"),
+        data,
+        [
+            QbColumn(
+                "Existing Dimension",
+                ExistingQbDimension(
+                    "https://example.org/dimensions/existing_dimension",
+                    arbitrary_rdf=[
+                        TripleFragment(RDFS.label, "Existing Dimension Component")
+                    ],
+                ),
+            ),
+            QbColumn(
+                "Value",
+                QbObservationValue(
+                    NewQbMeasure("Some Measure"), NewQbUnit("Some Unit")
+                ),
+            ),
+            QbColumn(
+                "Other Value",
+                QbObservationValue(
+                    NewQbMeasure("Some Other Measure"), NewQbUnit("Some Other Unit")
+                ),
+            ),
+        ],
+    )
+
+    dsd_helper = DsdToRdfModelsHelper(cube, UriHelper(cube))
+    dataset = dsd_helper._generate_qb_dataset_dsd_definitions()
+    graph = dataset.to_graph(Graph())
```

Comment:
We don't need this graph or the assertion do we?

Reply:
sorted